Services/CO.CDP.Organisation.WebApi/UseCase/CreateOrganisationJoinRequestUseCase.cs

```diff
@@ -42,6 +42,14 @@ public async Task<OrganisationJoinRequest> Execute((Guid organisationId, CreateO
         var person = await personRepository.Find(command.createOrganisationJoinRequestCommand.PersonId)
                            ?? throw new UnknownPersonException($"Unknown person {command.createOrganisationJoinRequestCommand.PersonId}.");
 
+        var alreadyAdded = await CheckPersonAlreadyAdded(organisation, person);
+
+        if (alreadyAdded)
+        {
+            throw new PersonAlreadyAddedToOrganisationException(
+                $"Person {command.createOrganisationJoinRequestCommand.PersonId} already added to organisation {command.organisationId}.");
+        }
```

Comment:
All these added lines seem to belong together. They accomplish one thing - guard that the person isn't already added. Why not replace all these lines with a single `async GuardPersonIsNotAlreadyAdded(organisation, person)`?

`CheckPersonAlreadyAdded` could be just renamed to `GuardPersonIsNotAlreadyAdded` and throw exception instead of returning a bool. 

Outcome: less code, and code in the public method remains on the same level of abstraction.

Reply:
@jakzal Thanks for the feedback. Done 👍